### PR TITLE
Military service episodes may have a nil end date. 

### PR DESCRIPTION
**Pull request by patrickvinograd**
Treat as ending in the future for sorting purposes.

See https://github.com/department-of-veterans-affairs/vets.gov-team/issues/6297

### CODE REVIEW

File: app/models/emis_redis/military_information.rb [hunk 1]
```diff
@@ -185,7 +185,7 @@ def military_service_episodes
 
     def service_episodes_by_date
       @service_episodes_by_date ||= lambda do
-        military_service_episodes.sort_by(&:end_date).reverse
+        military_service_episodes.sort_by { |ep| ep.end_date || Time.zone.today + 3650 }.reverse
```
Comment (lihanli) [inline app/models/emis_redis/military_information.rb]: how about `DateTime::Infinity`?

Comment (patrickvinograd) [reply app/models/emis_redis/military_information.rb]: Playing around with that in rails console, it doesn't seem like it wants to sort `DateTime::Infinity.new` against other `Date` instances.

Comment (lihanli) [reply app/models/emis_redis/military_information.rb]: how about `Date::Infinity`?

Comment (patrickvinograd) [reply app/models/emis_redis/military_information.rb]: Same - and looks like Date::Infinity is deprecated per apidock...I'm not sure what the `::Infinity` class is supposed to be used for but it doesn't seem like it's this.

**Pull request merged**